packages/apps/reputation-oracle/server/src/modules/abuse/abuse.service.spec.ts

```diff
@@ -0,0 +1,495 @@
+  },
+  EscrowUtils: {
+    getEscrow: jest.fn(),
+  },
+}));
+
+describe('AbuseService', () => {
+  let abuseService: AbuseService;
+  let abuseSlackBot: AbuseSlackBot;
+  let abuseRepository: AbuseRepository;
+  let webhookOutgoingService: WebhookOutgoingService;
+  let reputationService: ReputationService;
+
+  const escrowAddress = faker.finance.ethereumAddress();
+  const chainId = faker.number.int({ max: 100000 });
+  const webhookUrl1 = faker.internet.url();
+  const webhookUrl2 = faker.internet.url();
+
+  beforeAll(async () => {
+    const moduleRef = await Test.createTestingModule({
+      providers: [
+        AbuseService,
+        ConfigService,
+        ServerConfigService,
+        {
+          provide: AbuseSlackBot,
+          useValue: createMock<AbuseSlackBot>(),
+        },
+        {
+          provide: Web3Service,
+          useValue: {
+            getSigner: jest.fn().mockReturnValue({
+              address: faker.finance.ethereumAddress(),
+              getNetwork: jest.fn().mockResolvedValue({ chainId }),
+            }),
+          },
+        },
+        { provide: AbuseRepository, useValue: createMock<AbuseRepository>() },
+        {
+          provide: ReputationService,
+          useValue: createMock<ReputationService>(),
+        },
+        {
+          provide: WebhookOutgoingService,
+          useValue: createMock<WebhookOutgoingService>(),
+        },
+      ],
+    }).compile();
+
+    abuseSlackBot = moduleRef.get<AbuseSlackBot>(AbuseSlackBot);
+    abuseService = moduleRef.get<AbuseService>(AbuseService);
+    abuseRepository = moduleRef.get<AbuseRepository>(AbuseRepository);
+    webhookOutgoingService = moduleRef.get<WebhookOutgoingService>(
+      WebhookOutgoingService,
+    );
+    reputationService = moduleRef.get<ReputationService>(ReputationService);
+  });
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('reportAbuse', () => {
+    it('should create a new abuse entity', async () => {
+      const userId = 1;
+
+      await abuseService.reportAbuse({
+        escrowAddress,
+        chainId,
+        userId,
+      });
+
+      expect(abuseRepository.createUnique).toHaveBeenCalledWith({
+        escrowAddress: escrowAddress,
+        chainId: chainId,
+        userId: userId,
+        retriesCount: 0,
+        status: AbuseStatus.PENDING,
+        waitUntil: expect.any(Date),
+      });
+    });
+  });
+
+  describe('processSlackInteraction', () => {
+    it('should send an Abuse Report Modal to Slack if the decision is accepted', async () => {
+      const abuseEntity = generateAbuseEntity({ status: AbuseStatus.NOTIFIED });
+
+      const dto = {
+        callback_id: abuseEntity.id,
+        chainId,
+        type: 'interactive_message',
+        actions: [{ value: AbuseDecision.ACCEPTED }],
+        trigger_id: faker.string.uuid(),
+        response_url: faker.internet.url(),
+      };
+
+      jest
+        .spyOn(abuseRepository, 'findOneById')
```

Comment:
You don't have to create new spy here, `abuseRepository` already mocked, so you should be able to mock resolved value w/o new spies